src/cascadia/TerminalApp/Pane.LayoutSizeNode.cpp

```diff
@@ -37,37 +37,10 @@ Pane::LayoutSizeNode& Pane::LayoutSizeNode::operator=(const LayoutSizeNode& othe
     size = other.size;
     isMinimumSize = other.isMinimumSize;
 
-    _AssignChildNode(firstChild, other.firstChild.get());
-    _AssignChildNode(secondChild, other.secondChild.get());
-    _AssignChildNode(nextFirstChild, other.nextFirstChild.get());
-    _AssignChildNode(nextSecondChild, other.nextSecondChild.get());
+    firstChild = other.firstChild ? std::make_unique<LayoutSizeNode>(*other.firstChild) : nullptr;
+    secondChild = other.secondChild ? std::make_unique<LayoutSizeNode>(*other.secondChild) : nullptr;
+    nextFirstChild = other.nextFirstChild ? std::make_unique<LayoutSizeNode>(*other.nextFirstChild) : nullptr;
+    nextSecondChild = other.nextSecondChild ? std::make_unique<LayoutSizeNode>(*other.nextSecondChild) : nullptr;

```

Comment:
This is definitely a lot shorter, but it's not more efficient by any means. It's actually significantly less efficient. Although I bet it doesn't really matter, since panes aren't really that widely used in the first place.

Reply:
Alright wait - is it more efficient or is it not? Leonard, you signed off on the central thesis of this PR but left a comment indicating that it didn't do anything of the sort?

Reply:
The new code is less performant because it recreates any heap allocations even if they already existed. The old code didn't do that. But it is more efficient in a way, because it's less code. If we want to optimize for that, then this PR is also a win. But if it's about performance then we should keep the old code and only fix the `.release()` call.

Using `std::unique_ptr`s to build a binary tree is not optimal in the first place however, so any sort of performance optimizations on this is a bit of a lost cause anyways. 😅
The correct way to store small binary heaps like that is to store them as a flat list. That is, at index 0 is the root, index 1 and 2 store its left and right splits, etc. There's no need for references or pointers or anything to traverse such a binary tree because the position of the 2 child nodes can be determined based on the position of the parent node. This style of binary tree representation is actually more common than the classic text book pointer-style tree we're using. It's described here: https://en.wikipedia.org/wiki/Binary_tree#Arrays